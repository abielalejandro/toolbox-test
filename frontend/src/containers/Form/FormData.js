import { useRef } from "react";
import PropTypes from "prop-types";
import { Form, Button, Container } from "react-bootstrap";

const FormData = ({ onSubmitForm }) => {
  const textRef = useRef("");

  const onSubmit = (evt) => {
    evt.preventDefault();
    onSubmitForm({ text: textRef.current.value });
    textRef.current.value = "";
  };

  return (
    <Container className="d-flex justify-content-center bg-danger">
      <Form onSubmit={onSubmit} className="d-flex justify-content-center">
        <Form.Control
          ref={textRef}
          required
          id="text"
          name="text"
          className="m-2"
          placeholder="Insertar texto"
        />
        <Button className="m-2" type="submit">
          Enviar
        </Button>
      </Form>
    </Container>
  );
};

FormData.defaultProps = {
  onSubmitForm: () => {},
};

FormData.propTypes = {
  onSubmitForm: PropTypes.func.isRequired,
};

export default FormData;
