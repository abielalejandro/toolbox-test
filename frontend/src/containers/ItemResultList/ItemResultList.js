import PropTypes from "prop-types";
import { Card, Container } from "react-bootstrap";

import ItemResult from "../../components/ItemResult/ItemResult";
const ItemResultList = ({ items }) => {
  return (
    <Container className="d-flex justify-content-center mt-5">
      <Card style={{ width: "60%" }}>
        <Card.Body>
          <Card.Title>Solicitudes</Card.Title>
          {items &&
            items.length > 0 &&
            items.map((item, i) => {
              return <ItemResult key={i} result={item} />;
            })}
        </Card.Body>
      </Card>
    </Container>
  );
};

ItemResultList.defaultProps = {
  items: [],
};

ItemResultList.propTypes = {
  items: PropTypes.arrayOf(
    PropTypes.shape({
      isPalindrome: PropTypes.bool.isRequired,
      text: PropTypes.string.isRequired,
    })
  ),
};

export default ItemResultList;
