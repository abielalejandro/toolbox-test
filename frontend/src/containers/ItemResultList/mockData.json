[
    { "isPalindrome": true, "origin": "ana", "text": "ana" },
    { "isPalindrome": false, "origin": "test", "text": "tset" }
]