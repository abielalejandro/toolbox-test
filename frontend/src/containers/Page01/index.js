import { useDispatch, useSelector } from "react-redux";
import FormData from "../Form/FormData";
import ItemResultList from "../ItemResultList/ItemResultList";
import { getTextData } from "../../redux/actions";

const Page01 = (props) => {
  const dispatch = useDispatch();
  const items = useSelector((state) => state.data);
  const onSubmit = (data) => {
    dispatch(getTextData(data));
  };

  return (
    <>
      <FormData onSubmitForm={onSubmit} />
      <ItemResultList items={items} />
    </>
  );
};

export default Page01;
