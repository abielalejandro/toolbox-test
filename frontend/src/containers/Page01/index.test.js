import React from "react";
import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import store from "../../redux/store";

import Page01 from "./index";
import FormData from "../Form/FormData";
import ItemResultList from "../ItemResultList/ItemResultList";

describe("<Page01 />", () => {
  it("Expect to not log errors in console", () => {
    // eslint-disable-next-line
    const spy = jest.spyOn(global.console, "error");
    const wrapper = shallow(
      <Provider store={store}>
        <Page01 />
      </Provider>
    );
    expect(wrapper).not.toBeNull();
    expect(spy).not.toHaveBeenCalled();
  });

  it("Should render and match the snapshot", () => {
    const wrapper = shallow(
      <Provider store={store}>
        <Page01 />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("renders page", () => {
    const wrapper = mount(
      <Provider store={store}>
        <Page01 />
      </Provider>
    );
    expect(wrapper.find(FormData)).toHaveLength(1);
    expect(wrapper.find(ItemResultList)).toHaveLength(1);
  });
});
