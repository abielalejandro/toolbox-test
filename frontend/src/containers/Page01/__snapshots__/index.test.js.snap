// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Page01 /> Should render and match the snapshot 1`] = `ShallowWrapper {}`;
